airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -460,13 +590,15 @@ def generate_new_table_name(self, is_intermediate: bool, suffix: str) -> str:
         tables_registry = self.tables_registry.union(self.local_registry)
         new_table_name = self.normalized_stream_name()
         if not is_intermediate and self.parent is None:
+            if suffix:
```

Comment:
what problem is this change solving? are we going to end up with users_ab1_properties_ab1 or something funky like that? 

maybe there's a unit test we could add to make it clear what the expected behavior should be here. to be clear, i don't have an opinion on what it should be, i just am struggling to understand what it is based on the code. i think unit tests will help document it.

Reply:
The suffix was used only for intermediate tables so far (tables ending with (_ab1, _ab2, etc)

In this PR, I added an extra non-intermediate table (materialized and exposed to the user) with a suffix `_scd` which is the history SCD type 2 table in addition to the usual table per stream (which is deduped SCD type 1)

So this change makes it possible to suffix non-intermediate tables.